README.md

```diff
@@ -162,11 +167,79 @@ If we already have a WMS GetMap URL, we can use it directly:
   const imageBlob4 = await legacyGetMapFromUrl(fullUrlWithWmsQueryString, ApiType.PROCESSING);
 ```
 
-# Running examples
+# Authentication for Processing API
+
+Requests to Processing API need to be authenticated.
+Authentication is done by requesting an authentication token and setting it as described in [Layers](#layers).
+The authentication token is retrieved by making a request to the `https://services.sentinel-hub.com/oauth/token` endpoint with the OAuth Client's id and secret.
+
+To get the OAuth Client's id and secret, a new OAuth Client must be created in [**User settings**](https://apps.sentinel-hub.com/dashboard/#/account/settings) on **Sentinel Hub Dashboard** under **OAuth clients**.
+OAuth Client's secret is shown only before the creation process is finished so be mindful to save it.
+
+In javascript requesting the authentication token can be done with builtin XMLHttpRequest:
+
+```javascript
+const clientId = /* OAuth Client's id, best to put it in .env file and use it from there */;
+const clientSecret = /* OAuth client's secret, best to put it in .env file and use it from there */;
+
+let xhr = new XMLHttpRequest();
+xhr.open("POST", 'https://services.sentinel-hub.com/oauth/token', true);
+xhr.setRequestHeader("Content-Type", "application/x-www-form-urlencoded");
+xhr.onreadystatechange = function() {
+  if (this.readyState === XMLHttpRequest.DONE && this.status === 200) {
+    console.log(xhr.responseText);
+  }
+}
+xhr.send("grant_type=client_credentials&client_id="+clientId+"&client_secret="+clientSecret);
+```
```

Comment:
Well written! May I suggest linking the [official docs](https://docs.sentinel-hub.com/api/latest/reference/#section/Authentication) and maybe using just the `XMLHttpRequest` example? The reason for a single example is that this is not this library's core functionality (it is well documented in SH docs), so it would be best to keep it short.

Reply:
Thanks. Changed and left just the `XMLHttpRequest` example.